File: src/main/java/model/matchlist/MatchList.java
package model.matchlist;

import java.util.ArrayList;

/**
 * @author Josiah Kendall
 *
 * The pojo wrapper for the match list object.
 * Associated Url:
 * https://oc1.api.riotgames.com/lol/match/v3/matchlists/by-account/{accountId}
 */
public class MatchList {

    private ArrayList<MatchSummary> matches;
    private int startIndex;
    private int endIndex;
    private int totalGames;

    public void setEndIndex(int endIndex) {
        this.endIndex = endIndex;
    }

    public void setMatches(ArrayList<MatchSummary> matches) {
        this.matches = matches;
    }

    public void setStartIndex(int startIndex) {
        this.startIndex = startIndex;
    }

    public void setTotalGames(int totalGames) {
        this.totalGames = totalGames;
    }

    public ArrayList<MatchSummary> getMatches() {
        return matches;
    }

    public int getEndIndex() {
        return endIndex;
    }

    public int getStartIndex() {
        return startIndex;
    }

    public int getTotalGames() {
        return totalGames;
    }
}

File: processing/src/main/java/model/matchlist/MatchList.kt
package model.matchlist

import java.util.ArrayList

/**
 * @author Josiah Kendall
 *
 * The pojo wrapper for the match list object.
 * Associated Url:
 * https://oc1.api.riotgames.com/lol/match/v3/matchlists/by-account/{accountId}
 */
class MatchList {
    lateinit var matches : ArrayList<MatchSummary>
    var startIndex : Int = 0
    var endIndex: Int = 0
    var totalGames : Int = 0

    constructor()
    constructor(matches :ArrayList<MatchSummary>,
                startIndex: Int,
                endIndex: Int ,
                totalGames: Int ) {
        this.matches = matches
        this.startIndex = startIndex
        this.endIndex = endIndex
        this.totalGames = totalGames
    }
}
